### DISCUSSION TITLE
How to access the address of an object implementing the __buffer__ interface

### DISCUSSION THREAD

**Discussion by GuillaumeLeclerc** [micropython/micropython#11229] [Using MicroPython]
Hello,

I would love to have access to the address of an array/bytesarray. without having to enter viper. This would be super useful to quickly write/read something using the `mem32` array and reinterpret bytes -> ints etc...

Any easy way to do that ?

So far this is the "easieast" way I managed to do it but it's not super clean:

```python
import micropython
from machine import mem32
from array import array

data = bytearray(100)

@micropython.viper
def extract_ptr_inner(d, arr):
    p = ptr32(d)
    arr2 = ptr32(arr)
    arr2[0] = int(p)

def extract_ptr(data):
    arr = array('L', [0])
    extract_ptr_inner(data, arr)
    return arr[0]

my_ptr = extract_ptr(data)
mem32[my_ptr] = 17
print(data[0:4])

```

**Comment by dlech** [comment]
Did you try `uctypes`?

**Reply by peterhinch** [reply]
Yes - specifically [uctypes.addressof](http://docs.micropython.org/en/latest/library/uctypes.html#uctypes.addressof).

**Comment by GuillaumeLeclerc** [comment]
Looks great thanks!!
